.github/workflows/CI-pr.yml

```
@@ -288,3 +288,69 @@ jobs:
 
       - name: Spell check
         uses: crate-ci/typos@master
+
+
+  just-hack:
+    name: check compilation for V2 features
+    runs-on: ${{ matrix.runner }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        runner:
+          - ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: "Fetch base branch"
+        shell: bash
+        run: git fetch origin $GITHUB_BASE_REF --depth 1
+
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
+        if: ${{ runner.os == 'Linux' }}
+        with:
+          make-default: true
+
+      - name: Get rust version from Cargo.toml
+        shell: bash
+        run: |
+          rust_version=$(yq -oy '.workspace.package.rust-version' Cargo.toml)
+          echo "rust_version=${rust_version}" >> $GITHUB_ENV
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: "${{ env.rust_version }}"
+
+      - name: Install sccache
+        uses: taiki-e/install-action@v2.33.28
+        with:
+          tool: sccache
+          checksum: true
```

```suggestion
      - uses: Swatinem/rust-cache@v2.7.0
```
Use this action instead of sccache for now.